beacon-chain/rpc/eth/v1/validator/validator.go

```diff
@@ -232,7 +235,51 @@ func (vs *Server) GetAggregateAttestation(ctx context.Context, req *v1.Aggregate
 
 // SubmitAggregateAndProofs verifies given aggregate and proofs and publishes them on appropriate gossipsub topic.
 func (vs *Server) SubmitAggregateAndProofs(ctx context.Context, req *v1.SubmitAggregateAndProofsRequest) (*emptypb.Empty, error) {
-	return nil, errors.New("Unimplemented")
+	ctx, span := trace.StartSpan(ctx, "validatorv1.GetAggregateAttestation")
+	defer span.End()
+
+	for _, agg := range req.Data {
+		if agg == nil || agg.Message == nil || agg.Message.Aggregate == nil || agg.Message.Aggregate.Data == nil {
+			return nil, status.Error(codes.InvalidArgument, "Signed aggregate request can't be nil")
+		}
+		sigLen := params.BeaconConfig().BLSSignatureLength
+		emptySig := make([]byte, sigLen)
+		if bytes.Equal(agg.Signature, emptySig) || bytes.Equal(agg.Message.SelectionProof, emptySig) {
+			return nil, status.Error(codes.InvalidArgument, "Signed signatures can't be zero hashes")
+		}
+		if len(agg.Signature) != sigLen || len(agg.Message.Aggregate.Signature) != sigLen {
+			return nil, status.Errorf(codes.InvalidArgument, "Incorrect signature length. Expected %d bytes", sigLen)
+		}
+
+		// As a preventive measure, a beacon node shouldn't broadcast an attestation whose slot is out of range.
+		if err := helpers.ValidateAttestationTime(agg.Message.Aggregate.Data.Slot,
+			vs.TimeFetcher.GenesisTime(), params.BeaconNetworkConfig().MaximumGossipClockDisparity); err != nil {
+			return nil, status.Error(codes.InvalidArgument, "Attestation slot is no longer valid from current time")
+		}
+	}
+
+	broadcastFailed := false
+	for _, agg := range req.Data {
+		v1alpha1Agg := migration.V1SignedAggregateAttAndProofToV1Alpha1(agg)
```

Comment:
It is required to map the aggregate to a `v1alpha1` type because there needs to be a 1-to-1 mapping between types and gossip topics. Since the `v1alpha1` type is already used, we have to keep it this way.